### DISCUSSION TITLE
val loss and accuracy not logging

### DISCUSSION THREAD

**Discussion by gkrampah** [Lightning-AI/pytorch-lightning#20314] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
this is my code and I am having difficulty getting the CVS logger and tensorboard loggers to output the validation loss and accuracies after each epoch. The training parameters are properly logged. I am using the latest version of lightning. Any assistance will be highly appreciated.  
    def _shared_step(self, batch):
        features, true_labels = batch["Sample"], batch["Label"].long()
        logits = self(features)

        loss = F.cross_entropy(logits, true_labels)
        predicted_labels = torch.argmax(logits, dim=1)
        batch_size = features.size(0)
        return loss, true_labels, predicted_labels, batch_size

    def training_step(self, batch, batch_idx):
        # use _shared_step
        loss, true_labels, predicted_labels, batch_size = self._shared_step(batch)

        train_acc = self.train_acc(predicted_labels, true_labels)

        self.log(
            "train_loss",
            loss,
            prog_bar=True,
            on_epoch=True,
            on_step=False,
            batch_size=batch_size,
            logger=True,
        )

        self.log(
            "train_acc",
            train_acc,
            prog_bar=True,
            on_epoch=True,
            on_step=False,
            batch_size=batch_size,
            logger=True,
        )

        return loss

    # def on_train_epoch_end(self):
    #     self.log(
    #         "train_acc",
    #         self.train_acc.compute(),
    #         prog_bar=True,
    #         on_epoch=True,
    #         logger=True,
    #     )
    #     self.train_acc.reset()

    def validation_step(self, batch, batch_idx):
        # use _shared_step
        loss, true_labels, predicted_labels, batch_size = self._shared_step(batch)
        # Computes validation accuracy on whole validation set
        val_acc = self.val_acc(predicted_labels, true_labels)

        # self.validation_loss_outputs.append(loss)
        # self.validation_accuracy_outputs.append(val_acc)

        self.log(
            "val_loss",
            loss,
            prog_bar=True,
            on_epoch=True,
            on_step=False,
            batch_size=batch_size,
            logger=True,
        )

        self.log(
            "val_acc",
            val_acc,
            prog_bar=True,
            on_epoch=True,
            on_step=False,
            batch_size=batch_size,
            logger=True,
        )

**Comment by evileleven** [comment]
I encountered the same issue, it seems that they do not show valid loss and accuracy anymore.
